packages/flutter/lib/src/material/button_theme.dart

```diff
@@ -79,12 +76,19 @@ class ButtonTheme extends InheritedWidget {
     EdgeInsetsGeometry padding,
     ShapeBorder shape,
     bool alignedDropdown = false,
+    Color buttonColor,
+    Color disabledColor,
+    Color highlightColor,
+    Color splashColor,
+    ColorScheme colorScheme = const ColorScheme.light(),
+    MaterialTapTargetSize materialTapTargetSize,
```

Comment:
I think we should just leave this as part of the regular theme. While it needs to exist for backwards compatibility, ideally we don't want new apps to find it easy to set to anything besides padded.

Reply:
It's already a property of the button classes and it needs to be funneled through ButtonTheme so that buttons can (just) depend on ButtonThemeData and not on the overall Theme.

I've made the ButtonThemeData.materialTapTargetSize property private. It's initialized by the constructor, which will usually be called by ThemeData, and returned by getMaterialTapTargetSize().